scripts/backup.sh

```
@@ -0,0 +1,244 @@
+    
+    if [ $deleted_count -gt 0 ]; then
+        local size_freed_human=$(numfmt --to=iec-i --suffix=B --format=%.1f "$total_size_freed")
+        log INFO "Cleanup completed: deleted $deleted_count files, freed $size_freed_human"
+    else
+        log INFO "Cleanup completed: no old backups found"
+    fi
+}
+
+# Generate backup report
+generate_backup_report() {
+    local backup_count=$(find "$BACKUP_PATH" -name "${DB_FILE%.db}_*.db.gz" -type f | wc -l)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)